### PR TITLE
r-bassi/searchbar

### LINKED ISSUE #17: Navbar issues
Creating new issue after merging PR from navbar:

1. incorporate the new Logo from figma into the navbar
2. add user navigation, based on the figma
3. add ability to click outside of burger popup to close the popup and when clicking on a nav link

### PR DESCRIPTION
**Pull request by r-bassi**
closes #17

### CODE REVIEW

**Comment by JoshLuong** [comment]
please delete the package lock :)

File: trippo/package.json [hunk 1]
```diff
@@ -3,20 +3,29 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@freakycoder/react-native-bounceable": "^0.2.5",
```
Comment (JoshLuong) [inline trippo/package.json:6]: Do we need all of these new dependencies? Are they being used?

File: trippo/package.json [hunk 2]
```diff
@@ -50,8 +59,16 @@
     ]
   },
   "devDependencies": {
+    "@babel/cli": "^7.14.5",
```
Comment (JoshLuong) [inline trippo/package.json]: Same qs ^

File: trippo/src/app/reducers/daySlice.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { createSlice } from "@reduxjs/toolkit";
-import { getMockTimeSlots } from 'app/states';
-import { Location } from './locationSlice';
+import { getMockTimeSlots } from "app/states";
```
Comment (JoshLuong) [inline trippo/src/app/reducers/daySlice.ts:2]: hmm if we're just committing a 'prettier' file that was changed, we should just let the other devs prettify them. We can also specify the path of the files we change to prettify, so we don't prettify the whole repo and push it in. If it's a hassle to take out of the PR, we can just leave it in!

File: trippo/src/components/navBar/Navbar.tsx [hunk 4]
```diff
@@ -24,6 +27,16 @@ const Navbar = (props: { history: any }) => {
   const classes = useStyles();
   const theme = useTheme();
   const [open, setOpen] = React.useState(false);
+  const [anchorEl, setAnchorEl] = React.useState(null);
+
+  const handleDropdownClick = (event: any) => {
+    setAnchorEl(event.currentTarget);
```
Comment (JoshLuong) [inline trippo/src/components/navBar/Navbar.tsx:33]: can we also use this for the left navigation pop out?

File: trippo/src/components/searchBar/Searchbar.styles.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import styled from "styled-components";
+
+
+export const Container = styled.div`
+  border-radius: 7px;
+  z-index: 1;
+  display: inline-block;
+  width: 45%;
+  position: relative;
```
Comment (JoshLuong) [inline trippo/src/components/searchBar/Searchbar.styles.ts:9]:
```suggestion
  position: relative;
    @media (max-width: 1030px) {
    width: 60%;
  }
```

File: trippo/src/components/searchBar/Searchbar.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import {DARK_BLUE} from "../../colors/colors";
+
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    padding: "2px 4px",
+    display: "flex",
+    alignItems: "center",
+    borderRadius: 25,
+    boxShadow: '0 3px 5px 2px rgba(0, 0, 0, .3)',
```
Comment (JoshLuong) [inline trippo/src/components/searchBar/Searchbar.tsx]:
```suggestion
    boxShadow: '0 3px 2px 0.5px rgba(0, 0, 0, .3)',
    height: "3em"
```

File: trippo/src/components/searchBar/Searchbar.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    
+  },
+  input: {
+    marginLeft: theme.spacing(1),
+    flex: 1
```
Comment (JoshLuong) [inline trippo/src/components/searchBar/Searchbar.tsx]:
```suggestion
    flex: 1,
    padding: "6px 0 6px 5px"
```

File: trippo/package.json (hunk 1 above)
Comment (r-bassi) [reply trippo/package.json:6]: Ah no sorry, that was when I was trying things out. Only the packages with imports are required!

File: trippo/package.json (hunk 2 above)
Comment (r-bassi) [reply trippo/package.json]: same as above

File: trippo/src/components/navBar/Navbar.tsx [hunk 8]
```diff
@@ -68,11 +81,32 @@ const Navbar = (props: { history: any }) => {
               <img alt="Trippo Logo" src="trippo.png" width="120"></img>
             </sc.LogoButton>
           </sc.Logo>
+          <IconButton
+            style={{ position: "absolute", right: 20 }}
+            onClick={handleDropdownClick}
+          >
+            <AccountCircle style={{ width: 35, height: 35, color: GREY }} />
```
Comment (JoshLuong) [inline trippo/src/components/navBar/Navbar.tsx:89]:
```suggestion
            <AccountCircle style={{ width: 35, height: 35, color: BLACK}} />
```
just to keep it consistent with the other ones :)

File: trippo/src/components/navBar/Navbar.tsx [hunk 9]
```diff
@@ -11,9 +11,12 @@ import IconButton from "@material-ui/core/IconButton";
 import MenuIcon from "@material-ui/icons/Menu";
 import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
 import ChevronRightIcon from "@material-ui/icons/ChevronRight";
+import AccountCircle from "@material-ui/icons/AccountCircle";
 import ListItem from "@material-ui/core/ListItem";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
 import { withRouter } from "react-router-dom";
-import { BLACK, WHITE } from "../../colors/colors";
+import { BLACK, GREY, WHITE } from "../../colors/colors";
```
Comment (JoshLuong) [inline trippo/src/components/navBar/Navbar.tsx:19]:
```suggestion
import { BLACK, WHITE } from "../../colors/colors";
```
if you decide to take out the grey icon

File: trippo/src/components/navBar/Navbar.tsx (hunk 4 above)
Comment (r-bassi) [reply trippo/src/components/navBar/Navbar.tsx:33]: Left nav is controlled by handleDrawer, but I do think left nav and user dropdown should be in separate components entirely, and I think I will refactor that when we need to start adding user account pages to keep things neater.

Comment (JoshLuong) [reply trippo/src/components/navBar/Navbar.tsx:33]: Sure, let's just add `handleDrawerClose();` to line 49, under `handleMenuClick`

File: trippo/src/app/reducers/daySlice.ts (hunk 3 above)
Comment (r-bassi) [reply trippo/src/app/reducers/daySlice.ts:2]: This was unintentional, maybe I hit something accidentally during merge of project_3 into this branch.

File: trippo/src/components/navBar/Navbar.tsx (hunk 9 above)
Comment (r-bassi) [reply trippo/src/components/navBar/Navbar.tsx:19]: yeah I tried with black before and again just now, it just looks too dark to me. I just color picked with console and the other icons are (#656464) and (#89898b). So I think keeping it GREY (#656464) is more consistent?

**Review by JoshLuong — APPROVED** [review]

**Pull request merged**